src/CMakeLists.txt

```diff
@@ -375,17 +394,96 @@ endif() # if (APP STREQUAL "bootloader")
 
 target_sources(${EXECUTABLE} PRIVATE
 	${BSP_FILES}
+    ${LIB_FILES}
+    ${THIRD_PARTY_FILES}
 	${APP_FILES}
 	${KEY_FILES}
 	${CMAKE_CURRENT_BINARY_DIR}/version.c)
 
 target_include_directories(${EXECUTABLE} PRIVATE
 	${BSP_INCLUDES}
+    ${LIB_INCLUDES}
+    ${APP_DIR_COMMON}
+    ${THIRD_PARTY_INCLUDES}
 	${APP_INCLUDES}
 	)
 
+if(NOT APP MATCHES "bootloader")
+# Include bm_core
+set(BM_CORE_LWIP_INCLUDES
+    ${LIB_DIR}/lwip
+    ${THIRD_PARTY_INCLUDES}
+)
+
+set(BM_CORE_FREERTOS_INCLUDES
+    ${THIRD_PARTY_INCLUDES}
+    ${LIB_DIR}/common
+    ${MEMFAULT_INCLUDES}
+    ${APP_DIR}
+    ${APP_DIR_COMMON}
+)
+
+include_directories("${LIB_DIR}/bm_integration")
+include(${LIB_DIR}/bm_core/cmake/bm_core.cmake)
+setup_bm_ip_stack(LWIP "${BM_CORE_LWIP_INCLUDES}")
+setup_bm_os(FREERTOS "${BM_CORE_FREERTOS_INCLUDES}")
+add_subdirectory(${LIB_DIR}/bm_core bmcore)
+target_compile_definitions(middleware PRIVATE ${APP_DEFINES})
+target_link_libraries(bmcore PUBLIC lwipcore)
+
+if (USE_MICROPYTHON STREQUAL 1)
+include(${LIB_DIR}/micropython/CMakeLists.txt)
+
+list(APPEND APP_DEFINES "USE_MICROPYTHON")
+list(APPEND APP_DEFINES "MICROPY_HEAP_SIZE=1024*64;${MICROPY_DEFINITIONS}")
+list(APPEND APP_LIBS micropython)
+list(APPEND APP_INCLUDES "${MICROPY_INCLUDE_DIRS}")
+list(APPEND APP_FILES
+        ${LIB_DIR}/micropython/micropython_freertos.c
+        ${LIB_DIR}/micropython/mphalport.c)
+
+target_compile_options(
+    micropython PRIVATE
+    -Wno-dangling-pointer -Wno-sign-compare
+    )
+
+set_source_files_properties(
+    ${LIB_DIR}/micropython/mphalport.c
+
+    DIRECTORY ${SRC_DIR}
+
+    PROPERTIES
+    COMPILE_FLAGS -Wno-unused-parameter
+    )
+
+set_source_files_properties(
+    ${LIB_DIR}/micropython/micropython_freertos.c
+
+    DIRECTORY ${SRC_DIR}
+
+    PROPERTIES
+    COMPILE_FLAGS -Wno-unused-parameter
+    )
+
+endif()
```

Comment:
nit: add a `USE_MICROPYTHON` comment since this is dozens of lines away from the if